### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by gdozot2**

### CODE REVIEW

File: types/components/information-control.d.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Map } from 'ol';
+import Control from "ol/control/Control";
+import InformationElement from '../types/information-element';
+export default class InformationControl extends Control {
+    map: Map;
+    information: InformationElement;
+    informationIsOpen: Boolean;
+    timout: any;
```
Comment (slecorne) [inline types/components/information-control.d.ts]:
```suggestion
    timeout: any;
```

File: src/styles/popup-information.css [hunk 2]
```diff
@@ -0,0 +1,49 @@
+.custom-popup-element {
```
Comment (slecorne) [inline src/styles/popup-information.css:1]: we probably need theming for css or at least dark mode

File: src/components/drawer.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      }),
+    });
+    this.map.addLayer(this.vector);
+    this.modify = new Modify({ source: this.source });
+    this.map.addInteraction(this.modify);
+    this.addInteraction();
+  }
+
+  setupInteraction() {
+    if ((this.onlyOneDraw && this.nbDraw < 1) || !this.onlyOneDraw) {
```
Comment (slecorne) [inline src/components/drawer.ts]: we probably need to disallow drawing in some cases

File: src/components/geojson-loader.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  vectorLayer: VectorLayer<VectorSource> = new VectorLayer();
+
+  constructor(map: Map, url: string) {
+    this.map = map;
+    fetch(url)
+      .then((response) => {
+        return response.json();
+      })
+      .then((json) => {
+        this.data = json;
```
Comment (slecorne) [inline src/components/geojson-loader.ts]: probably no need to store it separately locally

File: src/components/geolocation-center.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import View from 'ol/View.js';
+import { Control } from 'ol/control'
+import Geolocation from 'ol/Geolocation';
+
+export default class GeolocationCenter extends Control {
+    map:Map;
+    view:View;
+    geolocaliseElement: Geolocation | undefined;
+
+    constructor(map:Map ,view:View, geolociliseElement:Geolocation | undefined) {
```
Comment (slecorne) [inline src/components/geolocation-center.ts]: remove `map` and  `view` from constructor and code to use `getMap` and `getView` from `Control`. See https://openlayers.org/en/latest/examples/custom-controls.html

File: src/components/geolocation-marker.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+import Point from 'ol/geom/Point';
+import VectorLayer from 'ol/layer/Vector';
+import Vector from 'ol/source/Vector';
+
+export default class GeolocationMarker {
+    map: Map;
+    geolocation: Geolocation;
+
+    vectorLayer = new VectorLayer();
+    vectorSource = new Vector;
```
Comment (slecorne) [inline src/components/geolocation-marker.ts:17]: why not using a marker like https://github.com/openlayers/openlayers/blob/main/examples/geolocation-orientation.js

File: src/components/information-box.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    element.style.setProperty('--progress-width', '100%');
+    title.className = 'custom-popup-title';
+    titleText.className = 'custom-popup-title-text';
+    content.className = 'custom-popup-content';
+    progress.className = 'custom-progress-element'
+    title.appendChild(titleText);
+    title.appendChild(cross);
+    element.appendChild(title);
+    element.appendChild(content);
+    element.appendChild(progress);
```
Comment (slecorne) [inline src/components/information-box.ts]: probably easier to use lit for reactivity in those sub components see https://github.com/MediaComem/geocity-wc-map/pull/2

File: types/components/information-control.d.ts (hunk 1 above)
Comment (gdozot2) [reply types/components/information-control.d.ts]: done

File: src/components/geojson-loader.ts (hunk 4 above)
Comment (gdozot2) [reply src/components/geojson-loader.ts]: Done

File: src/components/geolocation-center.ts (hunk 5 above)
Comment (gdozot2) [reply src/components/geolocation-center.ts]: Right done

File: src/components/information-box.ts (hunk 7 above)
Comment (gdozot2) [reply src/components/information-box.ts]: Port done

File: src/components/drawer.ts (hunk 3 above)
Comment (gdozot2) [reply src/components/drawer.ts]: Solve

**Review by slecorne — APPROVED** [review]

**Pull request merged**